rskj-core/src/test/java/co/rsk/peg/federation/FederationFactoryTest.java

```diff
@@ -0,0 +1,206 @@
+    }
+
+    @Nested
+    @Tag("erpFeds")
+    class ErpFederationTests {
+        @BeforeEach
+        @Tag("erpFeds")
+        void setUp() {
+            emergencyKeys = bridgeConstants.getErpFedPubKeysList();
+            activationDelayValue = bridgeConstants.getErpFedActivationDelay();
+            activations = mock(ActivationConfig.ForBlock.class);
+        }
+
+        @Test
+        void differentBuilderMethods_return_differentFederations(){
+            Federation federationStandard = createStandardMultisigFederation();
+            ErpFederation federationNonStandard = createNonStandardErpFederation();
+            ErpFederation federationP2sh = createP2shErpFederation();
+
+            assertNotEquals(federationStandard.getFormatVersion(), federationNonStandard.getFormatVersion());
+            assertNotEquals(federationStandard.getFormatVersion(), federationP2sh.getFormatVersion());
+            assertNotEquals(federationNonStandard.getFormatVersion(), federationP2sh.getFormatVersion());
+        }
+
+        @Test
+        void differentNonStandardErpFederations_areNotEqualFeds_butHaveSameNonStandardFedFormat_testnet() {
+            bridgeConstants = BridgeTestNetConstants.getInstance();
+            networkParameters = bridgeConstants.getBtcParams();
+
+            int version;
+
+            // in testnet it should build non standard hardcoded fed
+            when(activations.isActive(ConsensusRule.RSKIP201)).thenReturn(true);
+            ErpFederation federationPostRSKIP201PreRSKIP284 = createNonStandardErpFederation();
+            version = federationPostRSKIP201PreRSKIP284.getFormatVersion();
+
+            assertEquals(NON_STANDARD_ERP_FEDERATION.getFormatVersion(), version);
+
+            // build non standard fed with csv unsigned big endian
+            when(activations.isActive(ConsensusRule.RSKIP284)).thenReturn(true);
+            ErpFederation federationPostRSKIP201PostRSKIP284PreRSKIP293 = createNonStandardErpFederation();
+            version = federationPostRSKIP201PostRSKIP284PreRSKIP293.getFormatVersion();
+
+            assertEquals(NON_STANDARD_ERP_FEDERATION.getFormatVersion(), version);
+            assertNotEquals(federationPostRSKIP201PreRSKIP284, federationPostRSKIP201PostRSKIP284PreRSKIP293);
+
+            // build non standard fed
+            when(activations.isActive(ConsensusRule.RSKIP293)).thenReturn(true);
+            ErpFederation federationPostRSKIP201PostRSKIP284PostRSKIP293 = createNonStandardErpFederation();
+            version = federationPostRSKIP201PostRSKIP284PostRSKIP293.getFormatVersion();
+
+            assertEquals(NON_STANDARD_ERP_FEDERATION.getFormatVersion(), version);
+            assertNotEquals(federationPostRSKIP201PreRSKIP284, federationPostRSKIP201PostRSKIP284PostRSKIP293);
+            assertNotEquals(federationPostRSKIP201PostRSKIP284PreRSKIP293, federationPostRSKIP201PostRSKIP284PostRSKIP293);
```

Comment:
```suggestion
            when(activations.isActive(ConsensusRule.RSKIP201)).thenReturn(true);
            ErpFederation federationPostRSKIP201 = createNonStandardErpFederation();
            version = federationPostRSKIP201.getFormatVersion();

            assertEquals(NON_STANDARD_ERP_FEDERATION.getFormatVersion(), version);

            // build non standard fed with csv unsigned big endian
            when(activations.isActive(ConsensusRule.RSKIP284)).thenReturn(true);
            ErpFederation federationPostRSKIP284 = createNonStandardErpFederation();
            version = federationPostRSKIP284.getFormatVersion();

            assertEquals(NON_STANDARD_ERP_FEDERATION.getFormatVersion(), version);
            assertNotEquals(federationPostRSKIP201, federationPostRSKIP284);

            // build non standard fed
            when(activations.isActive(ConsensusRule.RSKIP293)).thenReturn(true);
            ErpFederation federationPostRSKIP293 = createNonStandardErpFederation();
            version = federationPostRSKIP293.getFormatVersion();

            assertEquals(NON_STANDARD_ERP_FEDERATION.getFormatVersion(), version);
            assertNotEquals(federationPostRSKIP201, federationPostRSKIP293);
            assertNotEquals(federationPostRSKIP284, federationPostRSKIP293);
```

What do you think about renaming those variables in this way? We can assume that for an N rskip, lower rskip are active, and vice versa.

Reply:
good idea, is clearer

Reply:
Better now